src/ref_serializer.rs

```diff
@@ -1,32 +1,35 @@
 use crate::serializer::Serializer;
 use crate::{AsRefBareItem, RefBareItem, SFVResult};
-use std::marker::PhantomData;
 
 /// Serializes `Item` field value components incrementally.
 /// ```
 /// use sfv::RefItemSerializer;
 ///
-/// let mut serialized_item = String::new();
-/// let serializer = RefItemSerializer::new(&mut serialized_item);
-/// serializer
-/// .bare_item(11)
-/// .unwrap()
-/// .parameter("foo", true)
-/// .unwrap();
+/// let serialized_item = RefItemSerializer::new()
+///     .bare_item(11)
+///     .unwrap()
+///     .parameter("foo", true)
+///     .unwrap()
+///     .finish();
 /// assert_eq!(serialized_item, "11;foo");
 /// ```
 #[derive(Debug)]
-pub struct RefItemSerializer<'a> {
-    pub buffer: &'a mut String,
+pub struct RefItemSerializer {
+    buffer: String,
 }
 
-impl<'a> RefItemSerializer<'a> {
-    pub fn new(buffer: &'a mut String) -> Self {
```

Comment:
One of the main purposes of RefItemSerializer was to avoid allocations.
So we could allocate a 100 byte buffer, and no allocations would happen when serializing (unless we go over).

How about a different approach - `RefItemSerializer::new` takes ownership of the buffer instead?

Reply:
We could take ownership of the buffer, yes. Would you expect us to clear it first, though? Otherwise, we need to think about the problem from #131.

Reply:
In that case we might as well continue to accept a mutable reference. I will update this PR accordingly.

Reply:
Updated so that the caller can decide whether to provide a buffer or use a fresh one.